monorail.py

```diff
@@ -1,16 +1,40 @@
 from oauth2client.client import GoogleCredentials
 import googleapiclient.discovery
+import google.auth.compute_engine
+import google.auth.impersonated_credentials
 
 DISCOVERY_URL = (
     'https://monorail-prod.appspot.com/_ah/api/discovery/v1/apis/'
     '{api}/{apiVersion}/rest')
 
+# https://cs.chromium.org/chromium/infra/go/src/infra/monorail/monorail.go?l=10&rcl=201f157c7ebc2be34ebeaa03ea47de2d8f4a8233
+MONORAIL_AUTH_SCOPE = 'https://www.googleapis.com/auth/userinfo.email'
+MONORAIL_SERVICE_ACCOUNT = 'monorail-api@ecosystem-infra.iam.gserviceaccount.com'
+# Local fallback.
 MONORAIL_KEY = 'monorail-key.json'
 
+_credentials = None
 
-def query(q):
-    credentials = GoogleCredentials.from_stream(MONORAIL_KEY)
 
+def _get_credentials():
+    global _credentials
+    if _credentials:
+        return _credentials
```

Comment:
https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials has "short lived" in the URL so presumably the credentials will expire? How to deal with that?

Reply:
We *might* need to refresh the tokens (the `Credentials` class has a method for that). I'll double check tomorrow to see if it's still working.

Reply:
It's still working today. Looks like we don't need to explicitly refresh the tokens.